scripts/install

```diff
@@ -1,41 +1,93 @@
-sudo yunohost app checkurl $domain$path -a hextris
-if [[ ! $? -eq 0 ]]; then
-    exit 1
-fi
+#=================================================
+# GENERIC START
+#=================================================
+# IMPORT GENERIC HELPERS
+#=================================================
+
+source /usr/share/yunohost/helpers
+
+#=================================================
+# MANAGE SCRIPT FAILURE
+#=================================================
+
+# Exit if an error occurs during the execution of the script
+ynh_abort_if_errors
+
+#=================================================
+# RETRIEVE ARGUMENTS FROM THE MANIFEST
+#=================================================
+
+domain=$YNH_APP_ARG_DOMAIN
+path_url=$YNH_APP_ARG_PATH
+is_public=$YNH_APP_ARG_IS_PUBLIC
+
+app=$YNH_APP_INSTANCE_NAME
+
+#=================================================
+# CHECK IF THE APP CAN BE INSTALLED WITH THESE ARGS
+#=================================================
+
+final_path=/var/www/$app
+test ! -e "$final_path" || ynh_die "This path already contains a folder"
+
+# Normalize the url path syntax
+path_url=$(ynh_normalize_url_path $path_url)
 
+# Check web path availability
+ynh_webpath_available $domain $path_url
+# Register (book) web path
+ynh_webpath_register $app $domain $path_url
 
-# Save app settings
-sudo yunohost app setting hextris is_public -v "$is_public"
+#=================================================
+# STORE SETTINGS FROM MANIFEST
+#=================================================
 
+ynh_app_setting_set $app domain $domain
+ynh_app_setting_set $app path $path_url
+ynh_app_setting_set $app is_public $is_public
+
+#=================================================
+# STANDARD MODIFICATIONS
+#=================================================
+# COPY SOURCES
+#=================================================
+
+ynh_app_setting_set $app final_path $final_path
 # Copy files to the right place
-final_path=/var/www/hextris
-sudo mkdir -p $final_path
-sudo cp -a ../sources/* $final_path
-
-# Set permissions
-sudo chown -R www-data: $final_path
-
-# Modify Nginx configuration file and copy it to Nginx conf directory
-sed -i "s@YNH_WWW_PATH@$path@g" ../conf/nginx.conf
-sed -i "s@YNH_WWW_ALIAS@$final_path/@g" ../conf/nginx.conf
-nginxconf=/etc/nginx/conf.d/$domain.d/hextris.conf
-sudo cp ../conf/nginx.conf $nginxconf
-sudo chown root: $nginxconf
-sudo chmod 600 $nginxconf
-
-if [ "$is_public" = "Yes" ];
+mkdir -p $final_path
+cp -a ../sources/. $final_path
+
+#=================================================
+# NGINX CONFIGURATION
+#=================================================
+
+# Create a dedicated nginx config
+ynh_add_nginx_config
+
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
+# SECURE FILES AND DIRECTORIES
+#=================================================
+
+# Set permissions to app files
+chown -R www-data: $final_path
```

Comment:
Shouldn't we create a dedicated user...?

Reply:
No, there's no php in this app. So, you can't use a dedicated user.

Reply:
At least, you actually don't need that, as it's only client side. Forget that 😉